Update pulseaudio module (#270677)
NixOS/nixpkgs: nixos/doc/manual/release-notes/rl-2405.section.md

- Programs written in [Nim](https://nim-lang.org/) are built with libraries selected by lockfiles.
  The `nimPackages` and `nim2Packages` sets have been removed.
  See https://nixos.org/manual/nixpkgs/unstable#nim for more information.

> - Add `homeMode` option to `hardware.pulseaudio` module to set permission of pulse user home directory to 755 when running in "systemWide" mode;
>   Fixes [Issue#114399](https://github.com/NixOS/nixpkgs/issues/114399).

Reviewer: ```suggestion
- The `hardware.pulseaudio` module now sets permission of pulse user home directory to 755 when running in "systemWide" mode. It fixes [issue 114399](https://github.com/NixOS/nixpkgs/issues/114399).
```
